ClosedXML_Tests/Excel/AutoFilters/AutoFilterTests.cs

```diff
@@ -122,11 +122,28 @@ public void CanCopyAutoFilterToNewSheetOnNewWorkbook()
 
                 using (var wb2 = new XLWorkbook(ms2))
                 {
-                    Assert.IsTrue(wb2.Worksheets.First().AutoFilter.Enabled);
+                    Assert.IsTrue(wb2.Worksheets.First().AutoFilter.IsEnabled);
                 }
             }
         }
 
+        [Test]
+        public void CannotAddAutoFilterOverExistingTable()
+        {
+            using var wb = new XLWorkbook();
+
+            var data = Enumerable.Range(1, 10).Select(i => new
+            {
+                Index = i,
+                String = $"String {i}"
+            });
+
+            var ws = wb.AddWorksheet();
+            ws.FirstCell().InsertTable(data);
+
+            Assert.Throws<ArgumentException>(() => ws.RangeUsed().SetAutoFilter());
```

Comment:
Getting `ArgumentException` here is a bit confusing as we do not provide any argument. `InvalidOperationException` seems to fit better.

Reply:
True. I'll change that. As I was coding this I thought about how we really need ClosedXML-specific exceptions.

Reply:
Then probably also `InvalidOperationException` in `XLWorksheet.CheckRangeNotOverlappingOtherEntities()`, right?

Reply:
Sure